### PR TITLE
Migrate to github actions and codecov

### PR DESCRIPTION
**Pull request by skjnldsv**
Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

### CODE REVIEW

**Comment by skjnldsv** [comment]
```
PHP Parse error:  syntax error, unexpected ')' in ./lib/PasswordValidator.php on line 90
Errors parsing ./lib/PasswordValidator.php

```

what :fearful:

**Comment by skjnldsv** [comment]
image

:scream:

**Comment by skjnldsv** [comment]
So much faster!! :rocket:

File: lib/PasswordValidator.php [hunk 1]
```diff
@@ -86,7 +86,7 @@ public function validate(string $password): void {
 		if (!empty($errors)) {
 			throw new HintException(
 				implode(' ', $errors),
-				implode(' ', $hints),
+				implode(' ', $hints)
```
Comment (nickvergessen) [inline lib/PasswordValidator.php:89]: Sorry 🙈

File: .github/workflows/app-code-check.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          tools: phpunit
+          extensions: mbstring, iconv, fileinfo, intl, sqlite, pdo_sqlite
+
+      - name: Checkout app
+        uses: actions/checkout@v2
+        with:
+          path: apps/${{ env.APP_NAME }}
+
+      - name: App code check
+        run: php occ app:check-code ${{ env.APP_NAME }}
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply .github/workflows/app-code-check.yml:55]: Entirely?

[the remark this answers is not in the document]
Comment (skjnldsv) [reply .github/workflows/app-code-check.yml]: Removed

**Review by nickvergessen — APPROVED** [review]

**Comment by nickvergessen** [comment]
btw might be personal preference, but codecov (with teh comments) gets quite spamy without a lot value. I would also not add something like it.

**Comment by skjnldsv** [comment]
> but codecov (with teh comments) gets quite spamy without a lot value. I would also not add something like it.

Agreed, it's just that people have been deprecating scrutinizer apparently, so I blindly followed the mass :shrug: 
Maybe I can just disable the comment :thinking:

**Pull request merged**